app/services/cfe/obtain_assessment_result_service.rb

```diff
@@ -0,0 +1,35 @@
+module CFE
+  class ObtainAssessmentResultService < BaseService
+    private
+
+    def cfe_url_path
+      "/assessments/#{@submission.assessment_id}"
+    end
+
+    def query_cfe_service
+      make_get_request
+    end
+
+    def make_get_request
+      conn.get cfe_url_path
+    end
+
+    def process_response
+      @submission.cfe_result = @response.body
+      if @response.status == 200
+        write_submission_history(@response, 'GET')
+        @submission.results_obtained!
+      else
+        @submission.fail!
+        raise_exception_error(
+          CFE::SubmissionError.new('Unsuccessful HTTP response code', @response.status),
+          'GET'
+        )
```

Comment:
That's results in a weird construction in `raise_exception_error` where another `CFE::SubmissionError` is created. Is there really not a simpler way to do this?

Reply:
Open to suggestions....

Reply:
I think you need two methods - one takes an error, and the other builds an error from the input:
```ruby
def raise_exeception_error(message:, backtrace: nil, http_method: 'POST', http_status: nil)
  @submission.submission_histories.create!(
    url: cfe_url,
    http_method: http_method,
    request_payload: request_body,
    http_response_status: http_status
    error_message: messge,
    error_backtrace: backtrace
  )
  raise CFE::SubmissionError.new(message, http_status)
end

def catch_and_record_exception(error, http_method = 'POST')
  raise_exeception_error(
    message: formatted_error_message(error),
    backtrace: error.backtrace&.join("\n"),
    http_method: http_method,
    http_status: error.respond_to?(:http_status) ? err.http_status : nil
  )
end
```
Then in instances where an error has been caught:
```ruby
rescue SomeError => e
  catch_and_record_exception(e)
end
```
And instances like this where we need to raise a new exception:
```ruby
 raise_exeception_error(message: 'Unsuccessful HTTP response code', http_status: @response.status)
```